gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -96,26 +93,15 @@ public GoogleCloudStorageGrpcWriteChannel(
 
   @Override
   public void setDirectUploadEnabled(boolean enableDirectUpload) {
-    // TODO(b/135137933): Figure out if we need to support non-resumable uploads.
```

Comment:
Why we don't need this TODO?

Reply:
I propose we not deal with non-resumable uploads in this connector; it could help with latency for small object writes, but I expect that's not the common HIC case.

Reply:
I think we still need direct upload, there are use cases for it, especially when clients use GCSIO library directly, not through HDFS API.

Also, it will be hard to use GCS channel by default if it throws exception when user enable direct upload.

Reply:
That was the behavior before my change; all I did was remove the TODO.

What does it mean to be a "direct upload"? The setter in the base class doesn't define what it means, and it's not clear to me it has anything to do with resumable uploads.

Reply:
I think direct upload implemented with `uploadType=multipart` parameter in JSON API:
https://cloud.google.com/storage/docs/json_api/v1/how-tos/upload

Essentially upload when you don't need to create a separate resumable upload session - it saves at least one round-trip.

If this is/will be supported in gRPC API too, then I think we should have this TODO and address it before GA.

Reply:
Ah, I see. Ok, I think it's valid to let the customer control that we could do a non-resumable upload this way. Updated the TODO about that, and made the method no longer throw.